src/System.CommandLine/Parsing/CommandResult.cs

```diff
@@ -53,28 +53,20 @@ public sealed class CommandResult : SymbolResult
 
                 if (Command.HasArguments)
                 {
-                    for (int i = 0; i < Command.Arguments.Count; i++)
-                    {
-                        SymbolResultTree.TryGetValue(Command.Arguments[i], out SymbolResult? parsedResult);
-                        cache.Add(Command.Arguments[i].Name, parsedResult);
-                    }
+                    Populate(cache, Command.Arguments);
```

Comment:
If I'm understanding this correctly, a call to `Populate` could be triggered by a custom parser before the complete command line has been parsed. Will symbols parsed afterward be findable by name?

Reply:
A custom parser gets access to `SymbolResult`-derived type, it does not provide a reference to `ParseResult`. The `CommandResult.GetValue<T>(string name)` is internal, so it's also impossible to call outside of S.CL before the parsing has been finished. I can move this method to `ParseResult` itself to avoid confusion.

Reply:
> it's also impossible to call outside of S.CL before the parsing has been finished

The specific call I was wondering about is calls to `ArgumentResult` from within a custom parse delegate.

Reply:
With my latest changes it's impossible